app.py

```diff
@@ -1998,6 +1998,82 @@ def unlink_globus_identities():
+            'globus_identity', 'google_email', 'github_username', 'slack_username', 'protocols_io_email',
+            'website', 'orcid', 'pm', 'pm_name', 'pm_email']
+
+# Get a list of all approved members (not including admins)
+def get_all_members_with_all_info():
+    members = list()
+    # Order members by ID DESC
+    wp_users = WPUser.query.order_by(WPUser.id.desc()).all()
+    for user in wp_users:
+        # Check if this target user is a member (capabilities will be empty dict if not member role)
+        capabilities = next((meta for meta in user.metas if (meta.meta_key == wp_db_table_prefix + 'capabilities') and ('member' in meta.meta_value)), {})
+        if capabilities:
+            # Use this check in case certain user doesn't have the connection info
+            if user.connection:
+                # Note user.connection returns a list of connections (should be only one though)
+                connection_data = user.connection[0]
+                # Also get the globus_user_id
+                wp_user_meta_globus_user_id = WPUserMeta.query.filter(WPUserMeta.user_id == user.id, WPUserMeta.meta_key.like('openid-connect-generic-subject-identity')).first()
+
+                # Construct a new member dict and add to the members list
+                member = {
+                    'globus_user_id': wp_user_meta_globus_user_id.meta_value,
+                    'first_name': connection_data.first_name,
+                    'last_name': connection_data.last_name,
+                    'organization': connection_data.organization,
+                    'component': connection_data.department,
+                    'role': connection_data.title,
+                    'bio': connection_data.bio,
+                    'email': user.user_email.lower()
+                }
+                keys = get_connection_keys()
+                obj = None
+                try:
+                    globus_user = next((meta for meta in user.metas if meta.meta_key == 'globus_username'), {'meta_value': ''})
+                    if globus_user is not None:
+                        member['globus_username'] = globus_user['meta_value'] if isinstance(globus_user, (dict)) else globus_user.meta_value
+                    for k in keys:
+                        if k == 'pm':
+                            obj = next((meta for meta in connection_data.metas if meta.meta_key == connection_meta_key_prefix + 'pm'), {'meta_value': ''})
+                            if obj is not None:
+                                member['pm'] = 'Yes' if obj.meta_value == '1' else 'No'
+                        else:
+                            obj = next((meta for meta in connection_data.metas if meta.meta_key == connection_meta_key_prefix + k), {'meta_value': ''})
+                            if obj is not None:
+                                member[k] = obj['meta_value'] if isinstance(obj, (dict)) else obj.meta_value
```

Comment:
Here we should overwrite the value of `component`, `role`, `organization` with the actual value that the user entered for `other_component`, `other_role`, `other_organization`. Otherwise we'll only see "Other" in those columns instead of the actual user provided value due to the handling at https://github.com/sennetconsortium/member-ui/blob/libpitt/member-info-4/app.py#L2065-L2070

Reply:
@yuanzhou Don't quite understand this request. Will ping you.